### PR TITLE
Linux 5.19 compat

### PR DESCRIPTION
**Pull request by behlendorf**
### Motivation and Context

Handle additional recent changes made to the Linux 5.19 kernel.
Plus a minor bit of cleanup.

### Description

5c84ebd72 autoconf: AC_MSG_CHECKING consistency
59d8f0488 Linux 5.19 compat: asm/fpu/internal.

### How Has This Been Tested?

Locally compiled with latest kernel from kernel.org.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [x] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Library ABI change (libzfs, libzfs\_core, libnvpair, libuutil and libzfsbootenv)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the OpenZFS [code style requirements](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [x] I have updated the documentation accordingly.
- [x] I have read the [**contributing** document](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/openzfs/zfs/tree/master/tests) to cover my changes.
- [ ] I have run the ZFS Test Suite with this change applied.
- [x] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

**Review by tonyhutter — APPROVED** [review]

**Review by AttilaFueloep — CHANGES_REQUESTED** [review]
Well `asm/fpu/internal.h` was empty since Linux 5.16 an it looks like it got removed in 5.19. Maybe you could adjust the commit message?

File: config/kernel-fpu.m4 [hunk 1]
```diff
@@ -27,18 +27,32 @@ AC_DEFUN([ZFS_AC_KERNEL_FPU_HEADER], [
 	],[
 		AC_DEFINE(HAVE_KERNEL_FPU_API_HEADER, 1,
 		    [kernel has asm/fpu/api.h])
-		AC_MSG_RESULT(asm/fpu/api.h)
+
+		ZFS_LINUX_TRY_COMPILE([
+			#include <linux/module.h>
+			#include <asm/fpu/internal.h>
```
Comment (AttilaFueloep) [inline config/kernel-fpu.m4:33]: Nit: Should be a tabs I guess, not four spaces.

File: config/kernel-fpu.m4 [hunk 2]
```diff
@@ -27,18 +27,32 @@ AC_DEFUN([ZFS_AC_KERNEL_FPU_HEADER], [
...
+		],[
+		],[
+			AC_DEFINE(HAVE_KERNEL_FPU_INTERNAL_HEADER, 1,
+			    [kernel has asm/internal/api.h])
```
Comment (AttilaFueloep) [inline config/kernel-fpu.m4:37]:
```suggestion
[kernel has asm/fpu/internal.h]
```

File: config/kernel-fpu.m4 (hunk 1 above)
Comment (behlendorf) [reply config/kernel-fpu.m4:33]: I'm not sure I follow, tabs are used here not spaces.

**Comment by behlendorf** [comment]
Thanks for the review.  I've gone ahead and updated the comments to clarify that as of 5.16 the header was effectively then finally removed in 5.19.

File: config/kernel-fpu.m4 (hunk 1 above)
Comment (AttilaFueloep) [reply config/kernel-fpu.m4:36]: Sorry, a github formatting issue then.  Line 32 is indented 4 spaces whereas line 24 above is indented 8 spaces as expected. So I assumed spaces were used.

**Review by AttilaFueloep — APPROVED** [review]

**Pull request closed without merging**